stetho/src/main/java/com/facebook/stetho/inspector/ChromeDevtoolsServer.java

```diff
@@ -132,7 +132,13 @@ private void handleRemoteRequest(JsonRpcPeer peer, JSONObject requestNode)
       response.result = result;
       response.error = error;
       JSONObject jsonObject = mObjectMapper.convertValue(response, JSONObject.class);
-      String responseString = jsonObject.toString();
+      String responseString;
+      try {
+        responseString = jsonObject.toString();
+      } catch (OutOfMemoryError e) {
+        // JSONStringer can cause an OOM when the Json to handle is too big.
+        throw new MessageHandlingException(e);
```

Comment:
This will cause us to close the socket, dismissing the inspector window.  Instead, convert the response yourself to an error (similar to how the `catch (JsonRpcException)` clause a few lines up does it)